types/src/validator_verifier.rs

```diff
@@ -159,54 +166,99 @@ impl ValidatorVerifier {
         }
     }
 
-    /// This function will successfully return when at least quorum_size signatures of known authors
-    /// are successfully verified. Also, an aggregated signature is considered invalid if any of the
-    /// attached signatures is invalid or it does not correspond to a known author. The latter is to
-    /// prevent malicious users from adding arbitrary content to the signature payload that would go
-    /// unnoticed.
-    pub fn verify_aggregated_struct_signature<T: CryptoHash + Serialize>(
+    pub fn generate_aggregated_signature(
         &self,
-        message: &T,
-        aggregated_signature: &BTreeMap<AccountAddress, bls12381::Signature>,
-    ) -> std::result::Result<(), VerifyError> {
-        self.check_num_of_signatures(aggregated_signature)?;
-        self.check_voting_power(aggregated_signature.keys())?;
-        for (author, signature) in aggregated_signature {
-            self.verify(*author, message, &signature.clone())?;
+        partial_signatures: &PartialSignatures,
+    ) -> Result<(AggregatedSignature, PublicKey), VerifyError> {
+        let validator_bitmask = self
+            .address_to_validator_info
+            .iter()
+            .map(|(address, _)| partial_signatures.signatures().contains_key(address))
+            .collect::<Vec<_>>();
+        // Perform an optimistic aggregation of the signatures.
+        let aggregated_sig = bls12381::Signature::aggregate(
+            partial_signatures.signatures().values().cloned().collect(),
+        )
+        .map_err(|_| VerifyError::FailedToAggregateKey)?;
```

Comment:
this is failed to aggregate signature instead of key

Reply:
You could use the term _signature share(s)_ to be consistent with our BLS documentation and some of the academic literature on multisignatures.